File: mcp/src/gs1belu_mpm_mcp/environment.py
"""Pure derivation of API host, OAuth token host, and OAuth ``audience`` from an
:class:`Environment` + API segment (``upload``/``download``).

There is no caller-supplied host or base URL anywhere in the public config surface
(see :mod:`gs1belu_mpm_mcp.config`): a fumbled ``audience`` is exactly the GS1 manuals'
``access_denied`` failure, so every host is derived here, once, instead of being a
value a consumer can get wrong. Mirrors ``Gs1BeluEnvironmentResolver`` in the C#/TS
SDKs (#9/#36) — the same *derive-not-accept* contract, re-expressed for Python.
"""

from __future__ import annotations

import enum

DEFAULT_API_VERSION = "v17"


class Environment(enum.Enum):
    UAT = "uat"
    PROD = "prod"


_API_HOSTS = {
    Environment.UAT: "api-uat.gs1belu.org",
    Environment.PROD: "api.gs1belu.org",
}

_TOKEN_HOSTS = {
    Environment.UAT: "login-uat.gs1belu.org",
    Environment.PROD: "login.gs1belu.org",
}


def api_host(environment: Environment) -> str:
    return _API_HOSTS[environment]


def token_host(environment: Environment) -> str:
    return _TOKEN_HOSTS[environment]


def audience(environment: Environment) -> str:
    """The OAuth ``audience`` claim, with the mandatory trailing slash baked in."""
    return f"https://{api_host(environment)}/"


def token_endpoint(environment: Environment) -> str:
    return f"https://{token_host(environment)}/oauth/token"


def base_url(environment: Environment, api_segment: str, api_version: str = DEFAULT_API_VERSION) -> str:
    return f"https://{api_host(environment)}/myproductmanager/{api_segment}/{api_version}"

File: sdks/typescript/packages/mpm-upload/src/environment.ts
/**
 * The GS1 Belgium & Luxembourg My Product Manager deployment to target. Selects the API host, the
 * OAuth token host, and the OAuth `audience` together — see {@link Gs1BeluEnvironmentResolver}.
 */
export type Gs1BeluEnvironment = "uat" | "prod";

const API_SEGMENT = "upload";

/**
 * Pure derivation of the API host, OAuth token host, and OAuth `audience` from a
 * {@link Gs1BeluEnvironment}. There is no caller-supplied host or base URL anywhere in the public
 * surface: a fumbled `audience` is exactly the GS1 manuals' `access_denied` failure, so it is
 * derived here, once, instead of being a field a consumer can get wrong.
 */
export const Gs1BeluEnvironmentResolver = {
  apiHost(environment: Gs1BeluEnvironment): string {
    return environment === "uat" ? "api-uat.gs1belu.org" : "api.gs1belu.org";
  },

  tokenHost(environment: Gs1BeluEnvironment): string {
    return environment === "uat" ? "login-uat.gs1belu.org" : "login.gs1belu.org";
  },

  /** The OAuth `audience` claim, with the mandatory trailing slash baked in. */
  audience(environment: Gs1BeluEnvironment): string {
    return `https://${this.apiHost(environment)}/`;
  },

  tokenEndpoint(environment: Gs1BeluEnvironment): string {
    return `https://${this.tokenHost(environment)}/oauth/token`;
  },

  baseUrl(environment: Gs1BeluEnvironment, apiVersion: string): string {
    return `https://${this.apiHost(environment)}/myproductmanager/${API_SEGMENT}/${apiVersion}`;
  },
};
